koku/api/query_params.py

```diff
@@ -425,25 +387,61 @@ def _set_access_ocp_all(self, provider, filter_key, access_key, raise_exception=
             elif access_list:
                 self.parameters["filter"][filter_key] = access_list
 
-    def _set_tag_keys(self):
+    def search_url_for_prefixes(self, prefix_list):
+        """Searches url and returns true if any prefix is found in a url."""
+        # the url_data property is percent encoded
+        # https://en.wikipedia.org/wiki/URL_encoding
+        # %5Btag == [tag:
+        # %5Bor%3Atag == [or:tag
+        decoded_url = unquote(self.url_data)
```

Comment:
Rather than unquoting this every time this method is called,  I think it makes more sense to store the unquoted value on the object. Or maybe inspect the `QueryDict` object directly. That may be more precise than sub string matching.

Reply:
I think I'm getting the bigger picture of how we're doing the parsing here. I'd have to see more examples of the actual values of the query string to see if `querystring_parser` is buying us much over `QueryDict().dict()` or other similar methods.

Reply:
Or if we don't need the values to be url encoded, just pass in the characters we care about as safe.

```python
@property
def url_data(self):
    """Get the url_data."""
    return self.request.GET.urlencode(safe="[]:")
```